common/flatpak-dir.c

```diff
@@ -7594,31 +7628,86 @@ flatpak_dir_install (FlatpakDir          *self,
         }
       else
         {
+          /* If the system-helper can find bindfs on the system, we try to make a pull using the bindfs by
+             mounting repo/tmp/flatpak-cache-XXXXXX directory. If this operation is unsupported or failed,
+             the pull is then tried with the fallback codepath. */
+
+          g_autofree gchar *repo_tmp_cache_dir = NULL;
+          g_autoptr(OstreeRepo) child_repo = NULL;
+          g_auto(GLnxLockFile) child_repo_lock = { 0, };
+          g_autoptr(GError) local_error = NULL;
+
+          if (!flatpak_dir_system_helper_call_create_pull_cache_dir (self,
+                                                                     installation ? installation : "",
+                                                                     &repo_tmp_cache_dir, cancellable, &local_error))
+            {
+              if (g_error_matches (local_error, G_DBUS_ERROR, G_DBUS_ERROR_UNKNOWN_METHOD) ||
+                  g_error_matches (local_error, G_DBUS_ERROR, G_DBUS_ERROR_NOT_SUPPORTED))
+                g_debug ("Cannot create pull cache dir in repo/tmp: %s", local_error->message);
+              else
+                g_warning ("Cannot create pull cache dir in repo/tmp: %s", local_error->message);
+            }
+          else
+            {
+              g_autoptr(GFile) repo_tmp_cache_dir_gfile = NULL;
+
+              repo_tmp_cache_dir_gfile = g_file_new_for_path (repo_tmp_cache_dir);
+              child_repo = flatpak_dir_create_child_repo (self, repo_tmp_cache_dir_gfile, &child_repo_lock, NULL, &local_error);
+              if (child_repo == NULL)
+                {
+                 /* Explicitly clearing the child_repo pointer closes any open FDs on the child_repo.
+                    This prevents EBUSY error failure during unmounting. */
+                  glnx_release_lock_file (&child_repo_lock);
+                  g_clear_object (&child_repo);
```

Comment:
By definition, this line cannot be needed, since it's within `if (child_repo == NULL) { ... }`.